x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/exchange/ExchangeServiceTests.java

```diff
@@ -412,10 +412,6 @@ public void sendResponse(TransportResponse transportResponse) throws IOException
                         }
                         ExchangeResponse newResp = new ExchangeResponse(page, origResp.finished());
                         origResp.decRef();
-                        while (origResp.hasReferences()) {
-                            newResp.incRef();
-                            origResp.decRef();
-                        }
```

Comment:
@dnhatn could you take a look at the ESQL changes in this PR? Esp. here, I'm not sure killing all references of the message was intended to be part of the test. (It validates ownership assumption, the channel should only be able to kill its "own" reference.)

Reply:
The loop ensures that newResp has the same number of references as origResp. This change should be okay, assuming the test passes.